pkg/sql/gcjob_test/gcjob_test_test: TestGCJobRetry failed

pkg/sql/gcjob_test/gcjob_test_test.TestGCJobRetry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10088479?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10088479?buildTab=artifacts#/) on master @ [a65bca232dfde1862e5aff343e7b1f3a379d6cd6](https://github.com/cockroachdb/cockroach/commits/a65bca232dfde1862e5aff343e7b1f3a379d6cd6):


```
=== RUN   TestGCJobRetry
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/4d64db87b2aa0e9fccbce189bc4b8745/logTestGCJobRetry3203294588
    test_log_scope.go:79: use -show-logs to present logs inline
    sql_runner.go:103: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:116
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:311
        	            				pkg/sql/gcjob_test/gcjob_test_test_test/pkg/sql/gcjob_test/gc_job_test.go:309
        	Error:      	Received unexpected error:
        	            	query 'SELECT running_status FROM crdb_internal.jobs WHERE job_id = 865358649662537729': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:314
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:116
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:311
        	            	  | pkg/sql/gcjob_test/gcjob_test_test_test.TestGCJobRetry
        	            	  | 	pkg/sql/gcjob_test/gcjob_test_test_test/pkg/sql/gcjob_test/gc_job_test.go:309
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) query 'SELECT running_status FROM crdb_internal.jobs WHERE job_id = 865358649662537729': expected:
        	            	  | waiting for MVCC GC
        	            	  | got:
        	            	  | NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/4d64db87b2aa0e9fccbce189bc4b8745/logTestGCJobRetry3203294588
--- FAIL: TestGCJobRetry (58.50s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGCJobRetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27926

Confirmed that the error is still happening:
```
❯ ./dev test --stress pkg/sql/gcjob_test -f=TestGCJobRetry
...
1789 runs so far, 0 failures, over 6m10s

--- FAIL: TestGCJobRetry (46.66s)
    test_log_scope.go:167: test logs captured to: /tmp/_tmp/cdaf2ca040f808db47a8fe919624dfd1/logTestGCJobRetry197048104
    test_log_scope.go:81: use -show-logs to present logs inline
    sql_runner.go:104:
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:117
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:312
        	            				pkg/sql/gcjob_test/gcjob_test_test_test/pkg/sql/gcjob_test/gc_job_test.go:309
        	Error:      	Received unexpected error:
        	            	query 'SELECT running_status FROM crdb_internal.jobs WHERE job_id = 880389839475113985': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:315
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:117
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:312
        	            	  | pkg/sql/gcjob_test/gcjob_test_test_test.TestGCJobRetry
        	            	  | 	pkg/sql/gcjob_test/gcjob_test_test_test/pkg/sql/gcjob_test/gc_job_test.go:309
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_arm64.s:1172
        	            	Wraps: (2) query 'SELECT running_status FROM crdb_internal.jobs WHERE job_id = 880389839475113985': expected:
        	            	  | waiting for MVCC GC
        	            	  |
        	            	  | got:
        	            	  | NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:522: -- test log scope end --
FAIL
I230707 15:35:02.767444 1 (gostd) testmain.go:95  [-] 1  Test //pkg/sql/gcjob_test:gcjob_test_test exited with error code 1

ERROR: exit status 1

1807 runs completed, 1 failures, over 6m14s
context canceled
FAIL
```

--- a/pkg/sql/gcjob_test/gc_job_test.go
+++ b/pkg/sql/gcjob_test/gc_job_test.go
@@ -300,8 +300,8 @@ func TestGCJobRetry(t *testing.T) {
 	defer s.Stopper().Stop(ctx)
 	tdb := sqlutils.MakeSQLRunner(db)
 	tdb.Exec(t, "CREATE TABLE foo (i INT PRIMARY KEY)")
-	tdb.Exec(t, "ALTER TABLE foo CONFIGURE ZONE USING gc.ttlseconds = 1;")
-	tdb.Exec(t, "DROP TABLE foo CASCADE;")
+	tdb.Exec(t, "INSERT INTO foo SELECT generate_series(1, 1000)")
+	tdb.Exec(t, "DROP TABLE foo")
 	var jobID string
 	tdb.QueryRow(t, `
 SELECT job_id

